### PR TITLE
[css-borders-4][editorial] Moved logical border properties from CSS Logical 1 to CSS Borders 4

### PR DESCRIPTION
**Pull request by SebastianZ**
An issue in CSS Borders 4 asked whether we should move the logical border properties from CSS Logical 1 to that spec.

As we generally define logical properties and values in their dedicated specs, these days, I thought it would make sense to do that for them as well. So I went ahead and copied those definitions over and renamed "flow-relative" by "logical" as that's the term that's commonly used in newer specs.

I kept the definitions in CSS Logical 1 untouched for now, though, as CSS Borders 4 is meant to supersede those definitions. (Is there anything I have to do to indicate this?)

Sebastian

### CODE REVIEW

**Review by fantasai — CHANGES_REQUESTED** [review]
+1 to merging these in, but, I think we should try to integrate them into the spec a bit better, so that they don't feel so tacked on. One of our goals (mine anyway) is to make the logical properties eventually feel like 1st-class citizens alongside the physical ones. Probably should do the same in css-box-4 as well, fwiw.

The existing properties should indeed stay in css-logical-1 ; that spec scopes itself to defining the logical equivalents of all CSS2 properties, and expecting future topical specs to supersede and expand on their relevant logical properties.

**Comment by SebastianZ** [comment]
> +1 to merging these in, but, I think we should try to integrate them into the spec a bit better, so that they don't feel so tacked on. One of our goals (mine anyway) is to make the logical properties eventually feel like 1st-class citizens alongside the physical ones. Probably should do the same in css-box-4 as well, fwiw.

I tried to improve that by merging the definitions of both the physical and logical ones. Hope that's better now.

Note that I did not merge everything from CSS Backgrounds 3 in yet. I plan to do that once the spec. is stable enough.

Sebastian

File: css-borders-4/Overview.bs [hunk 1]
```diff
@@ -96,11 +89,239 @@ Line Colors: the 'border-color' properties</h3>
...
+	or
+	'border-inline-start' &amp; 'border-inline-end',
+	respectively,
+	both to the same style.
+
+	<!-- intentionally cannot set two sides independently;
+		see discussion in https://github.com/w3c/csswg-drafts/issues/3519 -->
+
+<h3 id="logical-shorthand-keyword">
+Four-Directional Shorthand Properties: the 'logical' keyword for 'border-width', 'border-style', and 'border-color'</h3>
```
Comment (fantasai) [inline css-borders-4/Overview.bs:231]: Drop this section, as this issue is being worked out in css-logical. Once that stabilizes, we can add relevant bits of spec text; but probably there will be a central definition that we can link to for these types of shorthands, not one that needs to be copied from spec to spec.

Comment (SebastianZ) [reply css-borders-4/Overview.bs]: Ok, removed.

**Review by fantasai — CHANGES_REQUESTED** [review]

File: css-borders-4/Overview.bs [hunk 2]
```diff
@@ -96,11 +89,174 @@ Line Colors: the 'border-color' properties</h3>
...
+
+	These two <a>shorthand properties</a> set the
+	'border-block-start-color' &amp; 'border-block-end-color'
+	and
+	'border-inline-start-color' &amp; 'border-inline-end-color',
+	respectively.
+	The first value represents the <a>start</a> edge color,
+	and the second value represents the <a>end</a> edge color.
+	If only one value is given,
+	it applies to both the <a>start</a> and <a>end</a> edges.
```
[the remark this answers is not in the document]
Comment (SebastianZ) [reply css-borders-4/Overview.bs]: Ok, done. Though Backgrounds 3 (and also Borders 4) still uses "edge" in other places.

File: css-borders-4/Overview.bs [hunk 3]
```diff
@@ -96,11 +89,174 @@ Line Colors: the 'border-color' properties</h3>
...
+	a missing bottom is the same as top,
+	and a missing right is also the same as top.
+	This is resolved individually for each list item.
+
+	The logical properties 'border-block-start-color', 'border-block-end-color',
+	'border-inline-start-color', and 'border-inline-end-color' correspond to the
+	'border-top-color', 'border-bottom-color', 'border-left-color',
+	and 'border-right-color' properties.
+	The mapping depends on the element's 'writing-mode', 'direction',
+	and 'text-orientation'.
```
[the remark this answers is not in the document]
Comment (SebastianZ) [reply css-borders-4/Overview.bs]: Done.

**Review by fantasai — APPROVED** [review]
Update the line in "changes" and otherwise it looks good!

**Pull request merged**